### DISCUSSION TITLE
MessageLoopProxy重定义问题

### DISCUSSION THREAD

**Discussion by 204065248** [netease-im/NIM_Duilib_Framework#335] [General]
## 问题现象
```
ui_components.lib(stdafx.obj) : error LNK2005: "public: void __thiscall nbase::MessageLoopProxy::PostTaskAndReplyRelay<void __cdecl(void),void __cdecl(void)>::Run(void)" (?Run@?$PostTaskAndReplyRelay@$$A6AXXZ$$A6AXXZ@MessageLoopProxy@nbase@@QAEXXZ) 已经在 pch.obj 中定义
```
请问这个是什么问题呢，之前代码重新编译就出现了。。。

**Comment by zhkkjun** [comment]
我也遇到这个问题了
把message_loop_proxy.h中最底下的
template<>
void MessageLoopProxy::PostTaskAndReplyRelay<void(), void()>::Run()
{
	std_task_();
	origin_loop_->PostTask(
		nbase::Bind(&PostTaskAndReplyRelay::RunReplyAndSelfDestruct, this));
}
注释掉试试

**Reply by reghtml** [reply]
注释掉确实可以编译通过了，但是为什么examples\basic 这个项目不需要注释呢，我复制出来一分同样的代码，能编译，自己创建一个新项目，就提示这个MessageLoopProxy重定义，  我不理解  ，调用的是同一个库啊

**Reply by 204065248** [reply]
不知道，网易的duilib说是在做新版，等新版出来再试试

**Reply by xuyunan** [reply]
老兄, 新版本预计什么时候能发布

**Reply by 204065248** [reply]
> 注释掉确实可以编译通过了，但是为什么examples\basic 这个项目不需要注释呢，我复制出来一分同样的代码，能编译，自己创建一个新项目，就提示这个MessageLoopProxy重定义， 我不理解 ，调用的是同一个库啊

![image](https://user-images.githubusercontent.com/17307843/190586494-07be42c8-31f8-4a6c-b1fa-daaa3a3895e7.png)
把这个关掉就好了
